### PR TITLE
unit template editing bug fix, vite config enhancement

### PR DESCRIPTION
**Pull request by happythenewsad**
## WHAT
1. allows UnitTemplates to be saved 
2. Add a minor config change to vite.json 
3. Convert spec file to react-testing-library

## WHY
1. expected behavior
2. To avoid the vite dev server from intermittently failing to serve assets [link](https://github.com/ElMassimo/vite_ruby/discussions/89)
3. react-testing-library migration is our long term strategy

## HOW
1. Use a safe navigation operator for `activityPackTime` variable 
2. config change 
3. .....

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Can-t-save-new-activity-pack-name-5a2dce28beba46e783280b803080ccd5

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? | yes
Have you deployed to Staging? | about to
Self-Review: Have you done an initial self-review of the code below on Github? |
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | (N/A or Yes)

### CODE REVIEW

**Review by dandrabik — APPROVED** [review]
:+1: LGTM

File: services/QuillLMS/client/app/bundles/Teacher/components/unitTemplate/__tests__/unitTemplate.test.jsx [hunk 1]
```diff
@@ -20,14 +20,34 @@ const unitTemplate = {
...
+      return []
+    });
+    const {getByText, asFragment} = render(
+      <UnitTemplate
+        returnToIndex={() => null}
+        unitTemplate={{unitTemplate, ...{time: null}}}
+      />
+    )
+    fireEvent.click(getByText(/Save/))
   });
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Teacher/components/unitTemplate/__tests__/unitTemplate.test.jsx]: It doesn't look to me like there are any actual assertions being made in this test. What are you expecting to happen when you click Save?

Comment (happythenewsad) [reply services/QuillLMS/client/app/bundles/Teacher/components/unitTemplate/__tests__/unitTemplate.test.jsx]: The assertion is implicit - if the patch is reverted, this spec will fail. If there's a better syntax, I'm happy to use it

File: services/QuillLMS/client/app/bundles/Teacher/components/unitTemplate/__tests__/unitTemplate.test.jsx [hunk 2]
```diff
@@ -20,14 +22,71 @@ const unitTemplate = {
...
+    // jest.mock(draftjs, 'EditorState', () => {
+    //   console.log("MOCKING EDITORSTATE\n\n........")
+    //   return jest.fn()
+    // })
+    // jest.spyOn(draftjs, 'genKey').mockImplementation(() => {
+    //   console.log("MOCKING genKey\n\n........")
+    //   return jest.fn()
+    // });
+
+    jest.spyOn(draftjs, 'ContentState').mockImplementation(() => {
```
Comment (happythenewsad) [inline services/QuillLMS/client/app/bundles/Teacher/components/unitTemplate/__tests__/unitTemplate.test.jsx]: draftjs creates nondeterministic attribute hashes in snapshots. enzyme ignored this, but react-testing-library does not. Which means that running `jest -u` on this spec file always succeeds, but `jest` always fails on a snapshot mismatch.

According to the jest docs:
https://jestjs.io/docs/snapshot-testing#property-matchers

... and this github issue in draftjs:
https://github.com/facebookarchive/draft-js/issues/702

mocking these two draftjs methods should make make the snapshot generation deterministic. However, my code here is not mocking `EditorState` and `ContentState`. The console logs are not printing.

File: services/QuillLMS/client/app/bundles/Teacher/components/unitTemplate/__tests__/unitTemplate.test.jsx [hunk 3]
```diff
@@ -20,14 +23,38 @@ const unitTemplate = {
...
+    jest.spyOn(unitTemplatesHelpers, 'validateUnitTemplateForm').mockImplementation(() => {
+      return []
+    });
+    const {getByText, asFragment} = render(
+      <UnitTemplate
+        returnToIndex={returnToIndexFunction}
+        unitTemplate={{unitTemplate, ...{time: null}}}
+      />
+    )
+    fireEvent.click(getByText(/Save/))
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Teacher/components/unitTemplate/__tests__/unitTemplate.test.jsx:59]: I get that an explicit assertion is tricky to write here, but this pattern still feels a little strange to me -- maybe just add a comment to clarify?

**Review by emilia-friedberg — APPROVED** [review]
Approving, with a note about preferring a comment on that test.

**Review by eadams17 — APPROVED** [review]

**Pull request merged**